resilience4j-spring/src/main/java/io/github/resilience4j/recovery/RecoveryMethod.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2019 Kyuhyen Hwang
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.github.resilience4j.recovery;
+
+import org.springframework.util.ReflectionUtils;
+import org.springframework.util.StringUtils;
+
+import java.lang.reflect.Method;
+import java.util.Arrays;
+
+public class RecoveryMethod {
+    private final Method recovery;
+    private final Object[] args;
+    private final Object target;
+
+    public RecoveryMethod(String recoveryMethodName, Method originalMethod, Object[] args, Object target) throws NoSuchMethodException, ClassCastException {
+        Class[] params = originalMethod.getParameterTypes();
+        Class[] recoveryParams = Arrays.copyOf(params, params.length + 1);
+        recoveryParams[params.length] = Throwable.class;
```

Comment:
Here I think you need to use Class<? extends Throwable> instead as this way you are forcing the user to specify the recovery method with Throwbale type parameter not specific exception type which is bad practice 
So you need to use do the following :
1- use instead Class<? extends Throwable>
2- check if the casue configured into recover method is assignable from it 

Second remark : 
you do not check the accessibility of the recovery method so you are assuming it is public which can be not the case all the time ,so you need to check if it is accessible otherwise make it accessible  :

// check isAccessible
 method.isAccessible();
// No then make it accessible 
ReflectionUtils.makeAccessible(method);